node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/editor/CursorImpl.java

```diff
@@ -0,0 +1,365 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+package com.yahoo.vespa.hosted.node.admin.task.util.editor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+/**
+ * @author hakon
+ */
+public class CursorImpl implements Cursor {
+    private final TextBuffer textBuffer;
+    private final Object unique = new Object();
+
+    private Position position;
+
+    /**
+     * Creates a cursor to a text buffer.
+     *
+     * WARNING: The text buffer MUST NOT be accessed outside this cursor. This cursor
+     * takes sole ownership of the text buffer.
+     *
+     * @param textBuffer the text buffer this cursor owns and operates on
+     */
+    CursorImpl(TextBuffer textBuffer) {
+        this.textBuffer = textBuffer;
+        position = textBuffer.getStartOfText();
+    }
+
+    @Override
+    public Position getPosition() {
+        return position;
+    }
+
+    @Override
+    public Mark createMark() {
+        return new Mark(position, textBuffer.getVersion(), unique);
+    }
+
+    @Override
+    public String getBufferText() {
+        return textBuffer.getString();
+    }
+
+    @Override
+    public String getLine() {
+        return textBuffer.getLine(position);
+    }
+
+    @Override
+    public String getPrefix() {
+        return textBuffer.getLinePrefix(position);
+    }
+
+    @Override
+    public String getSuffix() {
+        return textBuffer.getLineSuffix(position);
+    }
+
+    @Override
+    public String getTextTo(Mark mark) {
+        validateMark(mark);
+
+        Position start = mark.position();
+        Position end = position;
+        if (start.isAfter(end)) {
+            Position tmp = start;
+            start = end;
+            end = tmp;
+        }
```

Comment:
Agreed. Unfortunately, there's no such min&max methods on Comparable in base Java, so I added such utility (instead of requiring an additional library).

Reply:
Cool!